linodecli/plugins/k8s-alpha.py

```diff
@@ -0,0 +1,286 @@
+#   because we don't want to later delete resources from multiple clusters!
+
+    # Make application directory if it doesn't exist
+    appdir = safe_make_appdir("k8s-alpha")
+    # Make the terraform project directory if it doesn't exist
+    terrapath = os.path.join(appdir, parsed.name)
+    safe_mkdir(terrapath)
+
+    # Move to the directory
+    os.chdir(terrapath)
+
+    # Generate the terraform file
+    terrafile = open('cluster.tf', 'w')
+    terrafile.write(gen_terraform_file(context, parsed.name, hashname))
+    terrafile.close()
+
+    # Generate terraform args
+    terraform_args = gen_terraform_args(parsed)
+
+    # Run the Terraform commands
+    spcall(['terraform', 'workspace', 'new', parsed.name])
+    call_or_exit(['terraform', 'init'])
+    # TODO: Before running the apply delete any existing Linodes that would
+    # cause the apply to fail.
+    terraform_apply_command = ['terraform', 'apply'] + terraform_args
+    call_or_exit(terraform_apply_command)
+
+    # Merge and/or create kubeconfig for the new cluster.
+    # Also, activate the kubeconfig context.
+    safe_mkdir(os.path.expanduser("~/.kube"))
+
+    # We expect this to be the path to the generated kubeconfig file
+    kubeconfig_new = replace_kubeconfig_user(terrapath, parsed.name)
+    kubeconfig_existing = os.path.expanduser("~/.kube/config")
+    # Create a merged kubeconfig file and set the context
+    # First set up the KUBECONFIG env var so that `kubectl config view --flatten`
+    # gives us a new merged config
+    os.environ["KUBECONFIG"] = "{}:{}".format(kubeconfig_existing, kubeconfig_new)
+    tempfilepath = 'tempkubeconfig'
+    tempf = open(tempfilepath, 'w')
+    call_or_exit(['kubectl', 'config', 'view', '--flatten'], stdout=tempf)
+    tempf.close()
+    shutil.move(tempfilepath, kubeconfig_existing)
+
+    # Set the kubeconfig context to the new cluster
+    call_or_exit(['kubectl', 'config', 'use-context', '{}@{}'.format(parsed.name, parsed.name)])
+
+    # We're done! We have merged the user's kubeconfigs.
+    # So, the user should be able to run something like
+    # `kubectl get pods --all-namespaces`
+    # and see the Linode CSI, CCM, and ExternalDNS controllers
+
+def delete(args, context):
+    parser = argparse.ArgumentParser("{} create".format(plugin_name), add_help=True)
+    parser.add_argument('name', metavar='NAME', type=str,
+                        help="A name for the cluster.")
+    parsed, _ = parser.parse_known_args(args)
+
+    # Get the appdir path
+    appdir = safe_make_appdir("k8s-alpha")
+    terrapath = os.path.join(appdir, parsed.name)
+    # Move to the terraform directory
+    os.chdir(terrapath)
+    call_or_exit(['terraform', 'destroy'])
+
+    # TODO: Also delete all NodeBalancers and Volumes using the cluster prefix
+
+def quoted_string_or_bare_int(val):
+    if type(val) is int:
+        return val
+    elif type(val) is str:
+        return '"{}"'.format(val)
+    else:
+        return None
```

Comment:
it looks like this function is only used in string formatting below.  Returning `None` will put the string "None" (without quotes) into the resulting string - you probably want to `return ''` here.

Reply:
Great point